core/server/master/src/main/java/alluxio/master/file/meta/MountTable.java

```diff
@@ -288,16 +289,17 @@ public Resolution resolve(AlluxioURI uri) throws InvalidPathException {
       if (mountPoint != null) {
         MountInfo info = mMountTable.get(mountPoint);
         AlluxioURI ufsUri = info.getUfsUri();
-        UnderFileSystem ufs;
-        try {
-          ufs = mUfsManager.get(info.getMountId()).getUfs();
+        try (CloseableResource<UnderFileSystem> ufsClientResource =
+                 mUfsManager.get(info.getMountId()).acquireUfsClientResource()) {
+          UnderFileSystem ufs = ufsClientResource.get();
+          AlluxioURI resolvedUri = ufs.resolveUri(ufsUri, path.substring(mountPoint.length()));
+          // TODO(adit): Change resolution to store a closeable resource
+          return new Resolution(resolvedUri, ufs, info.getOptions().isShared(), info.getMountId());
```

Comment:
I think we need to address the TODO, otherwise someone can be using the ufs instance without it being tracked?

Reply:
that is true. i'm on it